truelearn/learning/_base.py

```diff
@@ -0,0 +1,565 @@
+                        f"The given constraint {expected_param_type} is not a class."
+                    )
+
+            if isinstance(expected_param_type, list):
+                # if match non of the type in the constraints
+                if not any(
+                    isinstance(param_value, param_type_unpacked)
+                    for param_type_unpacked in list(expected_param_type)
+                ):
+                    param_classname_expected = list(
+                        map(lambda cls: cls.__name__, expected_param_type)
+                    )
+                    raise TypeError(
+                        f"The {param_name} parameter of {self.__class__.__name__}"
+                        f" __init__ function must be one of the classes"
+                        f" in {param_classname_expected}."
+                        f" Got {param_value.__class__.__name__} instead."
+                    )
+            else:
+                if not isinstance(param_value, expected_param_type):
+                    raise TypeError(
+                        f"The {param_name} parameter of {self.__class__.__name__}"
+                        f" must be {expected_param_type.__name__}."
+                        f" Got {param_value.__class__.__name__} instead."
+                    )
+
+
+class InterestNoveltyKnowledgeBaseClassifier(BaseClassifier):
+    """A Base Classifier for KnowledgeClassifier, NoveltyClassifier \
+    and InterestClassifier.
+
+    It defines the necessary instance variables and
+    common methods to interact with the LearnerModel.
+    """
+
+    DEFAULT_CONTENT_SIGMA: Final[float] = 1e-9
+    DEFAULT_DRAW_PROBA_LOW: Final[float] = 1e-9
+    DEFAULT_DRAW_PROBA_HIGH: Final[float] = 0.999999999
+
+    _parameter_constraints: Dict[str, Any] = {
+        **BaseClassifier._parameter_constraints,
+        "learner_model": [LearnerModel, type(None)],
+        "threshold": float,
+        "init_skill": float,
+        "def_var": float,
+        "tau": float,
+        "beta": float,
+        "positive_only": bool,
+        "draw_proba_type": str,
+        "draw_proba_static": [float, type(None)],
+        "draw_proba_factor": float,
+    }
+
+    def __init__(
+        self,
+        *,
+        learner_model: Optional[LearnerModel],
+        threshold: float,
+        init_skill: float,
+        def_var: float,
+        beta: float,
+        tau: float,
+        positive_only: bool,
+        draw_proba_type: str,
+        draw_proba_static: Optional[float],
+        draw_proba_factor: float,
+    ) -> None:
+        """Init InterestNoveltyKnowledgeBaseClassifier object.
+
+        Args:
+            *:
+                Use to reject positional arguments.
+            learner_model:
+                A representation of the learner.
+            threshold:
+                A float that determines the prediction threshold.
+                When the predict is called, the classifier will return True iff
```

Comment:
when the predict method is called?

Reply:
Maybe I can remove all of these and just use "A float that determines the classification threshold". The "classifier threshold" is the technical term for this.